fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/rule/BaseMaterializedViewRewriteRule.java

```diff
@@ -146,6 +155,16 @@ public List<OptExpression> transform(OptExpression queryExpression, OptimizerCon
                         MaterializedViewMetricsRegistry.getInstance().getMetricsEntity(mvContext.getMv().getMvId());
                 mvEntity.increaseQueryMatchedCount(1L);
             }
+
+            // Do not try to enumerate all plans, it would take a lot of time
+            int limit = context.getSessionVariable().getCboMaterializedViewRewriteLimit();
```

Comment:
`getCboMaterializedViewRewriteLimit ` is to limit the final rewritten size of One Rule rather than the total rewritten mv size in the memo, it is expected? If true, we can add more comments in the SessionVariable.

Reply:
yes, it's for one rule but not for the whole query.

Reply:
clarify the comments